src/commcare_cloud/fab/deploy_diff.py

```diff
@@ -63,9 +64,16 @@ def get_diff_context(self):
             return context
 
         context["compare_url"] = self.url
+
+        if not self.repo.permissions:
+            # using unauthenticated API calls, skip diff creation to avoid hitting rate limits
+            print(color_warning("Diff generation skipped. Supply a Github token to see deploy diffs."))
```

Comment:
Maybe out of scope for this PR, but would it be possible to provide a github link to the difference? Like the links in the _Deploy History_ section of the [HQ system admin page](https://www.commcarehq.org/hq/admin/system/).

Reply:
That's the `compare_url` just above this, actually.

Reply:
Oh, nice. Looks like it's already included.